membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -214,13 +214,13 @@ object AccountDetails {
   }
   def mmaCategoryFrom(product: Product): String = product match {
     case _: Product.Paper => "subscriptions" // Paper includes GW 🤦‍
-    case _: Product.ZDigipack => "subscriptions"
-    case _: Product.SupporterPlus => "recurringSupport"
-    case _: Product.TierThree => "recurringSupport"
-    case _: Product.GuardianPatron => "subscriptions"
-    case _: Product.Contribution => "recurringSupport"
-    case _: Product.Membership => "membership"
-    case _ => product.name // fallback
+    case Product.Digipack => "subscriptions"
+    case Product.SupporterPlus => "recurringSupport"
+    case Product.TierThree => "recurringSupport"
+    case Product.GuardianPatron => "subscriptions"
+    case Product.Contribution => "recurringSupport"
+    case Product.Membership => "membership"
+    case _ => "subscriptions" // fallback - passing undefined value breaks manage
```

Comment:
this line means that all unknown products will be treated as subscriptions (although they will have already been filtered out earlier in the logic)

Reply:
I wonder whether this case ought to be handled in the client. Presumably the scenario where a product ends up with it's 'mmaCategory' property set to "subscriptions" by default is an error. The client (manage-frontend) uses the mmaCategory as a visual filter on the account overview, but also to mark selected products for specific functionality, like in the snappy named `augmentProductDetailWithDeliveryAddressChangeEffectiveDateForToday` function.

I guess at a higher level, what do we expect manage-frontend to render if its get's passed one of these products?

Reply:
good question - in actual fact the logic should filter them out before this line, but the compiler doesn't know that (any more since I removed the high level model)
I would have expected manage to just put all the unknown ones at the bottom, but I'd rather it put them in the wrong place than crashed completely.
I would prefer manage to either hide them completely, or more ideally print the subscription id and mark it as unknown in some way.

Reply:
`or more ideally print the subscription id and mark it as unknown in some way`

I like this idea